scheme/Makefile

```diff
@@ -8,6 +8,7 @@ SUBDIR += psa-iot
 SUBDIR += tpm-enacttrust
 SUBDIR += parsec-tpm
 SUBDIR += parsec-cca
+SUBDIR += cca-realm-provisioning
```

Comment:
@setrofim : As discussed only the `provisioning` will be a separate scheme, as it is coming from a different supply chain, the Verification will be taken care inside a combined CCA plugin, which verifies both Realm and Platform part together. Hence I chosen the name cca-realm-provisioning.

If I keep it just `cca-realm` then it will assume both Provisioning and Verifications are handled inside!

Reply:
Mmm ok, that would mean that the `StoreHandler` implementation would essentially be split/duplicated across two schemes, as the key formats need to be aligned across both schemes (which breaks the conceptual integrity of a scheme -- schems ought to be independent of each other).

Are we sure this is the right way forward? Would it not be better to implement as single `cca` scheme that handles everything in a single set of plugins, and then use configuration and/or policy to control whether the platform or realm parts, or both, are handled by a particular deployment?

Reply:
Sure, I thought about StoreHandler part, while implementing the provisioning scheme and came to conclusion that the internal `logic` will be implemented in the `common` set of functions and both the schemes will invoke the same internal implementation thus essentially avoiding duplication! This is the best way of approaching as the CCA Scheme StoreHandler needs to fetch in sequence the RefValues for Platform and Realm in sequence, however this scheme will only do that latter!

Reply:
Conceptually, a "scheme" is supposed to encompass the entire attestation "lifecycle", i.e. provisioning and attestion.  It doesn't make sense to have a scheme with only provisioning, after all, the sole purpose of provisioning endorsements is so that they would later be used in verification.

> the Verification will be taken care inside a combined CCA plugin, which verifies both Realm and Platform part together.

If that is the case, then this should be part of that combined plugin as well (going back to my intial suggestion of renaming this). When implementing the verification part of it, we can then re-use the commmon functionality between this and the `cca-ssd-platform` scheme.

On the other hand, if the goal is to modify `cca-ssd-platfom` scheme to optinally handle the realm part as well, then (it should probably be renamed, and) this should be part of that too. 

Basically, it doesn't make sense for provisioning to exist in a scheme of its own, as  the only purpose of provisioning is to enable verification. So if realm verification on its own is not a thing, and we can only have plaform and platform+realm, then we should have two schemes: `cca-ssd-platform` and `cca` (or `cca-combined`).

Reply:
> On the other hand, if the goal is to modify cca-ssd-platfom scheme to optinally handle the realm part as well, then (it should probably be renamed, and) this should be part of that too.

this looks like the "natural" choice: if there is realm ref-values, then we also appraise the realm, otherwise we only appraise the platform.

Reply:
> > On the other hand, if the goal is to modify cca-ssd-platfom scheme to optinally handle the realm part as well, then (it should probably be renamed, and) this should be part of that too.
> 
> this looks like the "natural" choice: if there is realm ref-values, then we also appraise the realm, otherwise we only appraise the platform.


Yes, the renaming of cca_ssd_platform to cca_scheme is on my work agenda.

However the Realm Provisioning cannot be part of that scheme.

It has a different suply chain. All I can accept is to rename this scheme as cca-realm, if both you and @thomas-fossati are ok. However, we agreed we do not need to implement `EvidenceHandler` interface for this scheme, as this will be done as part of CCA Scheme Verification!

Reply:
> It has a different suply chain. 

Why does this necessiate a different scheme? The difference is encapsulated in the media type, no? So the provisioning plugin for the combined scheme can differentiate on, and handle, both media types, and therefore both supply chains.

Reply:
Realm has a different "lifecycle"

Reply:
Realm provisioning has no bearing and no connection to cca-platform lifecycle, it is completely independent! Fusing them together will be incorrect!

Reply:
OK, but again, why does necessiate a different scheme? You do not need separate schemes to handle CoRIMs of different formats, or coming from different sources -- this is a matter of media type, not scheme.

Reply:
As long as realm and platform can be supplied in independent CoRIMs from independent provisioners, using the same media type (i.e., scheme) for both has no drawbacks that I can foresee.

Reply:
> using the same media type (i.e., scheme) 
 
media type != scheme. A scheme can support multiple media types. But yes, depending on how things are implemented, it may not even be necessary to define multiple media types (if the one media type defines the union of the entries from the different suppliers, and all entries are optional). 

I can see the value of separate media types (i.e CoRIM profiles) targeting different entities within the supply chain; those can still be handled within one scheme though.

My point is that the sole purpose of provisining endorsements is so that the are used in verification. The verifier determines what needs to be provisioned for a successful verification. So it does not make sense to have a "scheme" that only defines provisioning. That "scheme" will always be tied to some other scheme (the one that defienes verification), and so is not an indepent attestation scheme by itself. Allowing this would just confuse the definition of "scheme" as a concept within the project, and will make deployment configuration more complicated and error prone (when enabling plugins for a scheme, the deployer would then need to check if there are any dependent schemes that also need to be enabled, and that their versions are compatible, etc).

Reply:
Media types defines the payload container. CoRIM for CCA Platform, is a different container than a CoRIM for Realm.

Using the same media type for both is incorrect, isn't it?

Reply:
> Using the same media type for both is incorrect, isn't it?

It's not incorrect, as long as the same CoRIM profile takes care of both platform and realm.

Instead, if we want to have separate profiles, we will have two media types.

Reply:
Yes, they should be separate media types/profiles.

Reply:
Yes, CCA Platform and CCA Realm are two separate profiles and hence we will need two media types!

Reply:
> It's not incorrect, as long as the same CoRIM profile takes care of both platform and realm.

They technically _could_ be handled with a single profile. However, I belive the arguement is that they _should_ be handled by separate profiles, as they're expected to be privided by different entitied within the supply chain, and that would simplify the profile that each entity needs to understand.

Reply:
there are clear practical advantages to separating the two.  I was just noting that it's not "incorrect" per see.

I think we are all in agreement about "two media types for one scheme"